src/bindings/python/tests/test_graph/test_node_factory.py

```diff
@@ -92,3 +94,27 @@ def test_node_factory_validate_missing_arguments():
         pass
     else:
         raise AssertionError("Validation of missing arguments has unexpectedly passed.")
+
+def test_extension_added_from_library():
+    if platform == "win32":
+        library_path = "openvino_template_extension.dll"
+    else:
+        library_path = "libopenvino_template_extension.so"
+
+    factory = NodeFactory()
+    factory.add_extension(library_path)
+
+    data = ov.parameter([1, 2], dtype=np.float32)
+    identity = factory.create("Identity", data.outputs())
+    model = Model([identity], [data])
+    compiled = compile_model(model)
+    tensor = np.array([[3, 4]], dtype=np.float32)
+    result = compiled(tensor)
+
+    # TODO: There is an issue with life time of objects, free resources explicitly
+    # otherwise segfault will occur. Workaround: create factory as a global variable.
+    del compiled
+    del model
+    del identity
```

Comment:
Looks like I don't know how to hold the library until all the users disappear. @ilyachur, looks like this is a common problem of extensions, how can I solve it?

Reply:
I tried to hold `SOExtension` object in `rt_info` for the node created from that extension, but it didn't help presumably because the operation object itself needs to be destroyed before that RT info tag is destroyed, but real sequence is obviously reversed.

Reply:
I can try to temporarily hold the library in `ov::Model` destructor while all the nodes are being destroyed. But it won't help if someone holds a shared pointer to the op from the library somewhere else, outside the normal `ov::Model` graph, or in case when such a node is destroyed during the transformations -- too many places where we need to care about it.

Another solution is to collect them in a global static object, but it would mean extension libraries are never unloaded (not a big problem for "normal" applications though).

@jiwaszki, @ilya-lavrenov, @ilyachur, would appreciate your suggestions.